bulkExport/state-machine-definition.yaml

```diff
@@ -0,0 +1,120 @@
+              Retry:
+                - ErrorEquals: [ "States.ALL" ]
+              Next: waitForCrawler
+            waitForCrawler:
+              Type: Wait
+              Seconds: 60
+              Next: getCrawlerStatus
+            getCrawlerStatus:
+              Type: Task
+              Resource: !GetAtt getCrawlerStatus.Arn
+              Retry:
+                - ErrorEquals: [ "States.ALL" ]
+              Next: choiceOnCrawlerStatus
+            choiceOnCrawlerStatus:
+              Type: Choice
+              Choices:
+                - Variable: "$.executionParameters.crawlerStatus"
+                  StringEquals: 'succeeded'
+                  Next: startExportJob
+                - Variable: "$.executionParameters.crawlerStatus"
+                  StringEquals: 'running'
+                  Next: waitForCrawler
+                - Variable: "$.executionParameters.crawlerStatus"
+                  StringEquals: 'failed'
+                  Next: updateStatusToFailed
+            startExportJob:
+              Type: Task
+              Resource: !GetAtt startExportJob.Arn
+              Retry:
+                - ErrorEquals: [ "States.ALL" ]
+              Next: waitForExportJob
+            waitForExportJob:
+              Type: Wait
+              Seconds: 10
+              Next: getJobStatus
+            getJobStatus:
+              Type: Task
+              Resource: !GetAtt getJobStatus.Arn
+              Retry:
+                - ErrorEquals: [ "States.ALL" ]
+              Next: choiceOnJobStatus
+            choiceOnJobStatus:
+              Type: Choice
+              Choices:
+                - Variable: "$.executionParameters.isCanceled"
+                  BooleanEquals: true
+                  Next: cancelExportJob
+                - Variable: "$.executionParameters.glueJobRunStatus"
+                  StringEquals: 'SUCCEEDED'
+                  Next: updateStatusToCompleted
+                - Or:
+                  - Variable: "$.executionParameters.glueJobRunStatus"
+                    StringEquals: 'STARTING'
+                  - Variable: "$.executionParameters.glueJobRunStatus"
+                    StringEquals: 'RUNNING'
+                  Next: waitForExportJob
+                - Or:
+                  - Variable: "$.executionParameters.glueJobRunStatus"
+                    StringEquals: 'FAILED'
+                  - Variable: "$.executionParameters.glueJobRunStatus"
+                    StringEquals: 'TIMEOUT'
+                  - Variable: "$.executionParameters.glueJobRunStatus"
+                    # STOPPING and STOPPED can only occur here if the job was forcefully stopped with a Glue API call from outside the FHIR server, so we treat it as failure
+                    StringEquals: 'STOPPING'
+                  - Variable: "$.executionParameters.glueJobRunStatus"
+                    StringEquals: 'STOPPED'
+                  Next: updateStatusToFailed
+            cancelExportJob:
```

Comment:
In Glue the API that is used to "cancel" a job is called "stop". Let's use the same terminology as glue to be consistent and rename this to `stopExportJob`.

https://docs.aws.amazon.com/glue/latest/webapi/API_BatchStopJobRun.html

Reply:
done